kuma/api/v1/views.py

```diff
@@ -347,3 +351,23 @@ def get_user(request, username):
     data = {field: getattr(user, field) for field in fields}
     data["avatar_url"] = get_avatar_url(user)
     return JsonResponse(data)
+
+
+@api_view(["POST"])
+def create_subscription(request):
+    if not flag_is_active(request, "subscription"):
+        return Response(
+            "subscription flag not active for this user",
+            status=status.HTTP_403_FORBIDDEN,
+        )
+
+    try:
+        user = request.user
+        create_stripe_customer_and_subscription_for_user(
+            user, user.email, request.data.get("stripe_token", "")
+        )
+        return Response(None, status=status.HTTP_201_CREATED)
+
+    except stripe.error.StripeError:
+        raven_client.captureException()
+        return Response(None, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
```

Comment:
Perhaps it boils down to intimate familiarity with what kinds of errors you can get from `stripe`. 
But what I would do is to see what happens if you'd do `request.data.get("stripe_token", "") + 'junk'`. 

After all, 
```
error.py:class StripeError(Exception):
error.py:class APIError(StripeError):
error.py:class APIConnectionError(StripeError):
error.py:class StripeErrorWithParamCode(StripeError):
error.py:class CardError(StripeErrorWithParamCode):
error.py:class IdempotencyError(StripeError):
error.py:class InvalidRequestError(StripeErrorWithParamCode):
error.py:class AuthenticationError(StripeError):
error.py:class PermissionError(StripeError):
error.py:class RateLimitError(StripeError):
error.py:class SignatureVerificationError(StripeError):
```
Some of those could definitely only happen because of us and not something that a "bad" `stripe_token`  could cause.

Reply:
While a lot of these errors you listed there would happen due to our wrondoing, I think we're guarded against many of them with the system check.
If a client does curl an incorrect token, which leads to a stripe error, that I would very much consider a client error and the most likely source of errors here. I probably am biased towards 422 because of my Rails-days. Happy to change it to another 4xx though.

Reply:
My preference is that we start from basics and incrementally build up. 
Drop the `try: except:` here in Python. It'll give us Sentry errors which we will study and learn from. If patterns emerge and we see a lot of bogus hacking attempts, we'll whitelist those and turn those around on the client (with some 4xx error, I don't care which xx). 

On the client, we should be prepared. Put in something like this for a while in Python:
```python
import random
if random.random() > 0.5:
   raise WhateverError('the client should deal with this')
```

Then you'll know if the `.catch()` in the `SubscriptionForm` works as expected and feels fair for clients. Because it's only a matter of time till a `APIConnectionError` happens when our AWS cluster has a network hiccup to the Stripe server and that's the kind of error that means the user should be informed and not have doubt that they got charged.